src/main.js

```diff
@@ -107,10 +102,374 @@ if (WebGL.isWebGLAvailable()) {
 
 }
 
-window.addEventListener('resize', onWindowResize, false);
+function setupCamera(container) {
+    const camera = new THREE.PerspectiveCamera(75, container.offsetWidth / container.offsetHeight, 0.1, 1000);
+    camera.up.set(0, 0, 1);
+    camera.position.set(70, -40, 15);
+    camera.lookAt(new THREE.Vector3(0, 0, 0));
+    return camera;
+}
+
+function setupRenderer() {
+    const renderer = new THREE.WebGLRenderer({canvas: canvas});
+    renderer.setSize(container.offsetWidth, container.offsetHeight);
+    return renderer;
+}
+
+function setupControls() {
+    controls = new OrbitControls(camera, renderer.domElement);
+    controls.object.up.set(0, 0, 1);
+    controls.update();
+}
```

Comment:
These functions are only called once as part of the setup.

I'm tempted to say that if they're only called once, then maybe we don't need them in a function. But the functions seem familiar from you're other three.js project, and they also seem to be about setting up a bunch of boilerplate. If that's the case, I'd say it probably makes sense to encapsulate this type of thing in another file.

If it's custom code for this project, then maybe it makes more sense just being part of the setup function, i.e.:

```js
function setup() {
   
    // ...
    
    // setup renderer
    const renderer = new THREE.WebGLRenderer({canvas: canvas});
    renderer.setSize(container.offsetWidth, container.offsetHeight);

    // setup controls
    controls = new OrbitControls(camera, renderer.domElement);
    controls.object.up.set(0, 0, 1);
    controls.update();

    // add axes
    const axesHelper = new THREE.AxesHelper(500);
    scene.add(axesHelper);

    // ...

}
```